sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentOptions.cs

```diff
@@ -32,8 +32,15 @@ internal AnalyzeSentimentOptions(TextAnalyticsRequestOptions options)
         /// will contain the result of this analysis.
         /// </summary>
         /// <remarks>
-        /// This property only applies for <see cref="TextAnalyticsClientOptions.ServiceVersion.V3_1"/>, <see cref="TextAnalyticsClientOptions.ServiceVersion.V2022_05_01"/>, and up.
+        /// This property only applies for <see cref="TextAnalyticsClientOptions.ServiceVersion.V3_1"/>, <see cref="TextAnalyticsClientOptions.ServiceVersion.V2022_05_01"/>, and newer.
         /// </remarks>
         public bool? IncludeOpinionMining { get; set; }
+
+        internal new AnalyzeSentimentOptions CheckSupported(TextAnalyticsClientOptions.ServiceVersion version)
+        {
+            base.CheckSupported(version);
+            Validation.SupportsProperty(IncludeOpinionMining, nameof(IncludeOpinionMining), TextAnalyticsClientOptions.ServiceVersion.V3_1, version);
```

Comment:
Too bad we don't have a way to declare a min version as metadata and not take a perf hit.   I wonder if we should consider somehow organizing the version requirements in one place, but I don't have an idea for how to structure that in a way that wouldn't be a huge readability mess.

Reply:
In Cadl, we'll be able to generate that from the API description in a straightforward way.  The outstanding question I think is how we would want to use it.

Reply:
@tg-msft and I have chatted about that briefly, and one way might be to use Cadl `@added` or even `@removed` decorators and combine them with source generators (that way we can do the same thing in HLCs) to "transform" a internal? method into a public one that runs checks, then calls that internal? (maybe private) method - effectively a compile-time equivalent of decorators in TypeScript/newer JavaScript and Python.

Reply:
I've captured this in #30852 but is beyond the scope for this PR.